### ISSUE TITLE
Output predictions of segmentation task: "labelo"

### ISSUE THREAD

**Issue by wasd12345** [Microsoft/O-CNN#16]
When running in test mode to predict part labels of a new object and eventually visualize the output, what's the best way to get the label prediction out of your network? I see at the bottom of the segmentation prototxt the following commented out section:

```
# ##################
# # for test only
# layer {
#   name: "prob"
#   type: "Softmax"
#   bottom: "deconv2_ip"
#   top: "prob"
# }

# layer {
#   name: "labelo"
#   type: "Split"
#   bottom: "label"
#   top: "labelo"
# }

# # for test only
# ##################
```

When I uncomment this and run the network, I get output such as:
```
...
I0118 23:50:10.051928 20327 solver.cpp:237]     Train net output #119: labelo = -1
I0118 23:50:10.051933 20327 solver.cpp:237]     Train net output #120: labelo = -1
I0118 23:50:10.051939 20327 solver.cpp:237]     Train net output #121: labelo = 1
I0118 23:50:10.051944 20327 solver.cpp:237]     Train net output #122: labelo = -1
I0118 23:50:10.051964 20327 solver.cpp:237]     Train net output #123: labelo = -1
I0118 23:50:10.051970 20327 solver.cpp:237]     Train net output #124: labelo = -1
I0118 23:50:10.051975 20327 solver.cpp:237]     Train net output #125: labelo = -1
I0118 23:50:10.051990 20327 solver.cpp:237]     Train net output #126: labelo = -1
I0118 23:50:10.051995 20327 solver.cpp:237]     Train net output #127: labelo = -1
I0118 23:50:10.052000 20327 solver.cpp:237]     Train net output #128: labelo = -1
I0118 23:50:10.052008 20327 solver.cpp:237]     Train net output #129: loss = 5.72206e-06 (* 1 = 5.72206e-06 loss)
I0118 23:50:10.052047 20327 solver.cpp:237]     Train net output #130: prob = 5.71219e-06
I0118 23:50:10.052055 20327 solver.cpp:237]     Train net output #131: prob = 0.123834
I0118 23:50:10.052062 20327 solver.cpp:237]     Train net output #132: prob = 0.25759
I0118 23:50:10.052068 20327 solver.cpp:237]     Train net output #133: prob = 0.312887
I0118 23:50:10.052074 20327 solver.cpp:237]     Train net output #134: prob = 0.302314
I0118 23:50:10.052079 20327 solver.cpp:237]     Train net output #135: prob = 0.259468
I0118 23:50:10.052094 20327 solver.cpp:237]     Train net output #136: prob = 0.295625
I0118 23:50:10.052100 20327 solver.cpp:237]     Train net output #137: prob = 0.501078
I0118 23:50:10.052116 20327 solver.cpp:237]     Train net output #138: prob = 5.71219e-06
I0118 23:50:10.052122 20327 solver.cpp:237]     Train net output #139: prob = 0.123834
I0118 23:50:10.052129 20327 solver.cpp:237]     Train net output #140: prob = 0.25759
I0118 23:50:10.052145 20327 solver.cpp:237]     Train net output #141: prob = 0.312887
I0118 23:50:10.052150 20327 solver.cpp:237]     Train net output #142: prob = 0.302314
I0118 23:50:10.052155 20327 solver.cpp:237]     Train net output #143: prob = 0.259468
```

Do you provide any tools to conveniently take this "labelo" output and turn it back into the labeled 3D model? Or any utilities doing similar things to visualize or interpret the outputs?

Thanks.

**Comment by wasd12345** [comment]
Also, when I run in test mode, by running the following script:

`$CAFFE_ROOT/build/tools/caffe test -model segmentation_5.prototxt -weights snapshots/octree_iter_1000.caffemodel`

then even when the "labelo" and "prob" layers are uncommented nothing is actually printed out. So when doing test mode, how do I get out my predictions for the segmentation labels?

Thanks!

**Comment by wang-ps** [comment]
There are many ways to dump the output of neural network in the caffe framework.
I provided one approach (this method is not flexible enough, you can also google it for better ways).
I have updated the readme file. Please refer to the instructions in the section `
O-CNN for Shape retrieval` and `O-CNN for Shape Segmentation` for instructions

**Comment by wasd12345** [comment]
Re: Small bug in output saving code:

Using the command from your readme worked successfully:
`
$CAFFE_ROOT/build/tools/caffe-d test -model seg_myOCTREES.prototxt -weights snapshots/my_octrees__iter_1000.caffemodel -blob_prefix=feature/_myOctrees_test_ -save_seperately=false -iterations=10`

This creates 2 files:
_myOctrees_test_labelo.dat
_myOctrees_test_prob.dat
as the combined results over all N iterations. 

One small issue though is if you do -save_seperately=true to save each iteration separately, there is no .dat suffix appended to filename output.

**Comment by wang-ps** [comment]
Yes, it is my mistake, sorry.
I update the caffe.cpp and the readme. Please rebuild the code, then run according to the instructions.
